Mobile version

ommarz opened the pull request:
Start portfolio and mobile first project. 
Use of flexbox and mobile first approach.
Update README.md information related to the project.
Add .gitignore file
Add node_modules/ to .gitignore
Add linters
Add images directory, images for background, social networks, and menu.
Add index.html file.
Add web page content in index.html
Add stylesheet.css 
Add content style in stylesheet.css

Zuhaib042 reviewed (commented):
Hi @ommarz 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has two issues:
- [ ] Kindly consider setting up HTML and CSS linters in your repo.
- [ ] Kindly consider explaining what you have done in the pull request summary in detail.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

ommarz committed:
[Remove] app_screenshot.png murple_logo.png

Remove images not relevant to project.
-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[UPDATE] README.md

Update information relevant to the project
-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[UPDATE] MIT.md

Update information related to project license.
Edit APP TYPE to WEB PAGE

-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[ADD] index.html

Add index.html file
-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[ADD] stylesheet.css

Add stylesheet.css file
-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[ADD] .gitignore

ADD .gitignore
-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[Add] node_modules/ to .gitignore

-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[Remove] vscode/launch.json
-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[EDIT] README.md

Edit Usage command section
-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[EDIT] README.md

Edit format
-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[Update] Author Info

Update LinkedIn link
-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[Update] index.html

Add header, nav, headline section.
Add logo class, div for hamburger icon
Add content in headline

ommarz committed:
[Add] [Social Network]

Add images for social networks and background.
Edit index.html to add social networks links
-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[Add] [rel="noopener] index.html

(Linters fix)
-# How does the change address the issue?
-# What side effects does this change have?
-# Include a link to the ticket, if any.

ommarz committed:
[Add] [classes on index.html]

Improvements
To get better visualization of the elements

ommarz committed:
[Add] [.socials class] in index.html

Improvement in div

ommarz committed:
[Add] [css styling]

Improvements on styling to setup mobile web page
Styling using flex on .toolbar, .logo, .hamburger, .headline, .greeting,
.presentation, .description, .socials

Cabraham1 reviewed (changes requested):
# INVALID ❌ 

Hi @ommarz   

While you made a great effort in this project, unfortunately, I cannot proceed to review your code. 😓

## Invalid Code Review Request ❌

- [ ] Unfortunately, you have submitted a project that is has no [linters](https://github.com/microverseinc/linters-config) installed 

_**Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready**_ 🤝🙏

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

_**Reviewed By [Kanza Tahreem](https://github.com/KanzaTahreem)**_

ommarz committed:
[Add] [workflows directory and linters.yml]

Improvements to have linters in project
Add linters

Cabraham1 reviewed (changes requested):
# Required Changes :recycle:
![almost-there](https://user-images.githubusercontent.com/65860678/170491251-b1add8ba-b18f-4648-a893-20e510d957fd.gif)

## Hey @ommarz   :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

- [x] please follow this [Link](https://github.com/microverseinc/linters-config/tree/master/html-css) to install linters correctly

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers, and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ommarz committed:
[Update] [.github/workflows/ directory]

Install linters correctly
Remove ./workflows directory and add .github/workflows directory

ommarz committed:
[Edit] [stylesheet.css to meet linters]

To meet linters requirements

ommarz committed:
[Edit] [stylesheet.css to meet linters requiirements]

To meet linters

ommarz committed:
[Edit] [stylesheet.css syntaxis]

To meet linters' requirements

ommarz committed:
[Edit] [stylesheet.css sintaxis]

To meet linters' requirements

ommarz committed:
[Edit] [stylesheet.css sintax]

To meet linters' requirements

ommarz committed:
[Edit] [stylesheet.css sintax]

To meet linters' requirements

ommarz committed:
[ADD] [generic font in stylesheet.css]

To meet linters' requirements

ommarz committed:
[Edit] [sintax in stylesheet.css]

Meet linters' requirements

ommarz committed:
[Remove] [whitespaces in stylesheet.css]

To meet linters required.
[Add] indentation in line 78:1 to meet linters' req

ommarz committed:
[Edit] [whitespaces in stylesheet.css]

To meet linters' requirements

ommarz committed:
[Remove] [comments]

To meet linter's requirements

ommarz committed:
[Fix] [Webhint error message]

To meet linters requirements
Install linters and missing packages correctly

Qoosim commented on index.html:
```diff
@@ -0,0 +1,27 @@
+     <body>
+      <header class="header">   
+        <nav class="toolbar">
+          <p class="logo">Omar Gutierrez</p>
```
- [x] You're required to wrap the logo on your page with `<a>` tag. This is for three reasons: convention, semantics, and user experience(UX). [Check this link](https://ux.stackexchange.com/questions/81727/why-is-it-standard-for-a-website-logo-to-navigate-to-the-home-page) for more info. Kindly implement this requirement.

Qoosim commented on index.html:
```diff
@@ -0,0 +1,27 @@
+        <nav class="toolbar">
+          <p class="logo">Omar Gutierrez</p>
+          <div class="hamburger"><img src="./images/Hamburger.svg" alt="menu"></div>
+        </nav>
```
- [x] It is required that the hamburger icon on the right side of your page should be wrapped inside a `nav` tag. The `nav` tag is for semantics reason, and the right position for convention and user experience for mobile devices. The content of your header should look something similar to this :point_down: 

```
    <header>
        <a href="#">Your Logo</a>
        <nav>
           <>Your hamburger icon</>
        </nav>
    </header>
```

Qoosim commented on index.html:
```diff
@@ -0,0 +1,27 @@
+            <a href="https://www.linkedin.com/in/ommarz/" target="_blank" rel="noopener"><img src="./images/Disabled LinkedIn.svg" alt="linkedin" class="social-icon"></img></a>
+            <a href="https://github.com/ommarz" target="_blank" rel="noopener"><img src="./images/Disabled GitHub.svg" alt="github" class="social-icon"></img></a>
+            <a href="https://www.twitter.com/ommarz/" target="_blank" rel="noopener"><img src="./images/Disabled Twitter.svg" alt="twitter" class="social-icon"></img></a>
+          </div>
```
- [x] It is required to icons uses a `ul` tag to group your social icons, and `li` per icon; each icon has a link (`a` tag). Apart from being semantically correct, the `a` tag will let you connect your social networks with the icons, using href. Kindly implement this requirement.
- [x] Also, some icons are not showing. You might need to look into this too and fix.

### Your Design
![your-social-icons](https://github.com/ommarz/Personal_Portfolio/assets/26597432/37395587-9aa8-4803-8bb5-462d07baa100)

### Figma Design
![figma-social-icons](https://github.com/ommarz/Personal_Portfolio/assets/26597432/421952a8-1a2a-409c-96df-51b537e93775)

Qoosim commented on index.html:
```diff
@@ -0,0 +1,27 @@
+        <section class="headline">
+          <h1 class="greeting">Hey There. I'm Omar.</h1>
+          <h3 class="presentation">I'm a Software Developer</h3>
+          <p class="description">I will help you build a product, feature or website. Look through some of my work and experience! If you like what you see and have a project you need coded, don't hesistate to contact me</p>
```
- [x] It is required to stick to the figma design as close as possible. This includes, margin, padding, font, typography among others. Kindly reduce the space between the header and the headline on your page so as to match the figma design

### Your Design
![your-title-spacing](https://github.com/ommarz/Personal_Portfolio/assets/26597432/5706757c-50ce-4bf8-8a45-662a2726994d)

### Figma Design
![figma-title-spacing](https://github.com/ommarz/Personal_Portfolio/assets/26597432/56ed4717-44c0-4963-8231-07efd17d9fc3)

- [x] The same rule applies to the line-spacing between your main title, subtitle, and the paragraph. Kindly compare the two screenshots below and walk your design alongside the figma design

### Your Design
![your-line-spacing](https://github.com/ommarz/Personal_Portfolio/assets/26597432/507615d0-e8e0-451a-800a-806a001d63ff)

### Figma Design
![figma-line-spacing](https://github.com/ommarz/Personal_Portfolio/assets/26597432/02076db7-430e-4582-825c-2dcf9257fc5f)

Qoosim commented on stylesheet.css:
```diff
@@ -0,0 +1,74 @@
+  background-size: auto;
+  background-attachment: fixed;
+  background-position: right bottom;
+}
```
- [x] Kindly ensure that the background image used on this section is not fixed, in order not to affect other contents that would be added later on.
- [x] Also, you will need to adjust the position as to match how it appears on the figma design. The background image for this section on the Figma does not overlap the texts. But your design does. Kindly look into that and fix.

### Your Design
![your-bg-img](https://github.com/ommarz/Personal_Portfolio/assets/26597432/e9b5fec0-aec4-41c9-90b5-3ba5d9800884)

### Figma Design
![figma-bg-img](https://github.com/ommarz/Personal_Portfolio/assets/26597432/b6f96be8-0f98-4224-8440-ae775ea2c22c)

Qoosim commented on README.md:
```diff
@@ -60,180 +28,129 @@ After you're finished please remove all the comments and instructions!
+## 🛠 Built With `<a name="built-with"></a>`
 
-> Describe the tech stack and include only the relevant sections that apply to your project.
+### Tech Stack `<a name="tech-stack"></a>`
```
- [x] Your don't need all the `<a>` tags infront of each headings on your README file. Kindly remove them.

![your-readme](https://github.com/ommarz/Personal_Portfolio/assets/26597432/10c0c95e-70ea-4e78-bae0-cbb0774442b2)

Qoosim reviewed (changes requested):
#  *Status: Required Changes* :recycle: 

Hi @ommarz, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 

![](https://camo.githubusercontent.com/38ce8c8b213b08e2d9bb85a0b79702ad512f3910f803ec3d7917b4e0110c9044/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f7741366159414f336a4836526d55494259732f67697068792e676966)

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ommarz committed:
[Fix] [index.html] structure

Imrpovement in UX and to use tech industry standards
Wrap logo in <a> tag
Update header and nav tags

ommarz committed:
[Add] [social networks icons]

To meet template requirements

ommarz committed:
[Edit] [stylesheet.css] styling

To meet template requirements

ommarz committed:
[Edit] [index.html] indentation

[Add] Crete Round font
To meet template requirements

ommarz committed:
[Add] [fonts]

Improvements to meet template requirements

ommarz committed:
[Fix] [stylesheet.css] styling

To meet template requirements
Edit sizes, margins and flex position

ommarz committed:
[Remove] [README.md] no necessary <a> tags

To improve clean README.md

ommarz committed:
[Add] [DOCTYPE]

ommarz commented:
![Captura de pantalla 2023-06-08 164907](https://github.com/ommarz/Personal_Portfolio/assets/89239670/df8f0803-9343-4cab-b02a-b5ef0dec68d4)

Already made changes for the project to look as close as possible to the template, hopefully, you'll see it as the image above. It is a comparison of the project vs the template

amon-cofie commented on index.html:
```diff
@@ -0,0 +1,34 @@
+        <h3 class="presentation">I'm a Software Developer</h3>
+        <p class="description">I will help you build a product, feature or website. Look through some of my work and experience! If you like what you see and have a project you need coded, don't hesistate to contact me</p>
+          <ul class="socials">
+            <li><a href="https://github.com/ommarz" target="_blank" rel="noopener"><img src="./images/Disabled GitHub.svg" alt="github" class="social-icon"></img></a></li>
```
I see you put in a great effort to fix the issue that the previous reviewer pointed out but unfortunately, it still persists. However, I figured why this problem exists and it's the link to the image has a typo. In you images folder, the target file name is ```Disabled Github.svg``` but you have GitHub spelt with a capital H here. You can easily fix this issue by correcting the spelling error. The line might look something like this in the end: 

```sh
 <li><a href="https://github.com/ommarz" target="_blank" rel="noopener"><img src="./images/Disabled Github.svg" alt="github" class="social-icon"></img></a></li>
```

ommarz replied:
Thanks for noticing that, on my preview it didn't look like that

amon-cofie commented on stylesheet.css:
```diff
@@ -0,0 +1,150 @@
+  flex-direction: column;
+  margin: 0;
+  padding: 0;
+} */
```
[OPTIONAL] Please try as much as possible not to leave commented code in your codebase 💪. Commented code makes your codebase cluttered and hard to read. It's okay to leave comments as pointers to key parts of the code but those comments should not contain actual code 😉

ommarz replied:
Done

amon-cofie commented on stylesheet.css:
```diff
@@ -0,0 +1,150 @@
+  flex: none;
+  order: 3;
+  flex-grow: 0;
+}
```
Well done! 👍. You fixed the issue the previous reviewer pointed out. There's just one last thing to do and that is to set the ```list-style``` to none so that the default bullet style of the ```ul``` tag is removed as it is not consistent with the Figma design 😉. In the end your code would look something like this: 

```sh 
.socials {
  display: flex;
  flex-direction: row;
  align-self: flex-start;
  align-items: center;
  justify-content: space-between;
  padding: 0;
  margin: 0;
  width: 200px;
  height: 40px;
  flex: none;
  order: 3;
  flex-grow: 0;
  list-style: none;
}
```

Don't forget to remove commented code 💪

ommarz replied:
Done!

amon-cofie commented on README.md:
```diff
@@ -60,229 +31,180 @@ After you're finished please remove all the comments and instructions!
-
-This project is [MIT](./LICENSE) licensed.
+## 📝 License `<a name="license"></a>`
 
```
Just like the previous reviewer mentioned, please remove ````<a name="license"></a>```` 😅

amon-cofie commented on README.md:
```diff
@@ -60,229 +31,180 @@ After you're finished please remove all the comments and instructions!
 
-> Describe between 1-3 key features of the application.
+### Key Features `<a name="key-features"></a>`
 
```
Just like the previous reviewer mentioned, please remove ````<a name="key-features"></a>```` 😅

amon-cofie reviewed (changes requested):
Hi @ommarz , 👋 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

<img src="https://media0.giphy.com/media/VlqzeaRnhCvPLfrSh6/giphy.gif"/>


 ## Highlights

🚀 You have a descriptive pull request summary
🚀 You fixed most of the changes requested
🚀 You did really good on the styling
🚀 You used GitHub flow properly
🚀 You have no linter errors


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ommarz committed:
[Edit] [typo in github image link]

ommarz committed:
[Remove] [code comments in stylesheet.css]

ommarz committed:
[Add] [list-style in .socials]

Improvements on <ul> format

ommarz committed:
[Remove] [README.md pending <a> tags]

To improve cleaner code

ommarz commented:
Hi @amon-cofie everything has been fixed, thanks for your feedback! I want to know if is possible to merge the PR or if should I wait for the option for a new review. This is my third review already.

amejid reviewed (approved):
Hi @ommarz, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<img src="https://media1.giphy.com/media/YKdm91XONLTaw/giphy-downsized-medium.gif"/>

## Highlights
- Linter checks passing :heavy_check_mark: 
- Correct workflow :heavy_check_mark: 
- Good UI design :heavy_check_mark: 
- Great semantic structure :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
